test/e2e/metallb_test.go

```diff
@@ -0,0 +1,279 @@
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		t.Run("Basic installation", func(t *testing.T) {
+			t.Cleanup(func() { test.UninstallCuratedPackage(packagePrefix) })
+			test.InstallCuratedPackage(packageName, packagePrefix, "--kube-version=1.22")
+			err = WaitForPackageToBeInstalled(test, ctx, packagePrefix, 30*time.Second)
+			if err != nil {
+				t.Fatalf("waiting for metallb package to be installed: %s", err)
+			}
+			err = test.KubectlClient.WaitForDeployment(context.Background(),
+				test.Cluster(), "1m", "Available", "test-metallb-controller", namespace)
+			if err != nil {
+				t.Fatalf("waiting for metallb controller deployment to be available: %s", err)
+			}
+			err = WaitForDaemonset(test, ctx, "test-metallb-speaker", namespace, 2, 30*time.Second)
+			if err != nil {
+				t.Fatalf("waiting for metallb controller deployment to be available: %s", err)
+			}
+		})
+
+		t.Run("Address pool configuration", func(t *testing.T) {
+			address := "10.100.100.1"
+			t.Cleanup(func() { test.UninstallCuratedPackage(packagePrefix) })
+			test.CreateResource(ctx, fmt.Sprintf(
+				`
+apiVersion: packages.eks.amazonaws.com/v1alpha1
+kind: Package
+metadata:
+  name: test 
+  namespace: eksa-packages
+spec:
+  packageName: metallb
+  config: |
+    IPAddressPools:
+      - name: default
+        addresses:
+          - %s/32
+    L2Advertisements:
+      - IPAddressPools: 
+        - default
+`, address))
+			err = WaitForPackageToBeInstalled(test, ctx, packagePrefix, 30*time.Second)
+			if err != nil {
+				t.Fatalf("waiting for metallb package to be installed: %s", err)
+			}
+			err = test.KubectlClient.WaitForDeployment(context.Background(),
+				test.Cluster(), "1m", "Available", "test-metallb-controller", namespace)
+			if err != nil {
+				t.Fatalf("waiting for metallb controller deployment to be available: %s", err)
+			}
+			err = WaitForDaemonset(test, ctx, "test-metallb-speaker", namespace, 2, 30*time.Second)
+			if err != nil {
+				t.Fatalf("waiting for metallb speaker deployment to be available: %s", err)
+			}
+
+			expectedConfig := fmt.Sprintf(
+				`address-pools:
+  - name: default
+    addresses:
+      - %s/32
+    protocol: layer2
+`, address)
+			err = WaitForResource(
+				test,
+				ctx,
+				"configmap/test-metallb",
+				namespace,
+				"{.data.config}",
+				20*time.Second,
+				NoErrorPredicate,
+				StringMatchPredicate(expectedConfig),
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			t.Cleanup(func() {
+				test.KubectlClient.Delete(ctx, "service", "my-service", "default", test.Cluster().KubeconfigFile)
+			})
+			test.CreateResource(ctx, `
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  type: LoadBalancer
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+`)
+			err = WaitForResource(
+				test,
+				ctx,
+				"service/my-service",
+				"default",
+				"{.status.loadBalancer.ingress[0].ip}",
```

Comment:
This is brittle. I wonder if something using a function like [assert.JSONEq](https://pkg.go.dev/github.com/stretchr/testify/assert#JSONEq) could make this more flexible?